File: src/test/scala/com/github/kayjamlang/tests/expressions/array/ArrayExpressionTest.scala
package com.github.kayjamlang.tests.expressions.array

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.expressions.{ArrayExpression, ValueExpression}
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object ArrayExpressionTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("[1,2,3,4,5]"))
  }
}

class ArrayExpressionTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = ArrayExpressionTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[ArrayExpression], expression getClass)
    val arrayExpression = expression.asInstanceOf[ArrayExpression]
    assertEquals(5, arrayExpression.values size)
    for (i <- arrayExpression.values indices) {
      val value = arrayExpression.values apply i
      assertSame(classOf[ValueExpression], value getClass)
      val constantValue = value.asInstanceOf[ValueExpression]
      assertEquals(i + 1, constantValue.value)
    }
  }
}

File: src/test/java/com/github/kayjamlang/tests/expressions/array/ArrayExpressionTest.java
package com.github.kayjamlang.tests.expressions.array;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.expressions.ArrayExpression;
import com.github.kayjamlang.core.expressions.ValueExpression;
import com.github.kayjamlang.tests.TestsUtils;
import org.junit.BeforeClass;
import org.junit.Test;


import static org.junit.Assert.*;

public class ArrayExpressionTest {

    @Test
    public void test() throws Exception {
        Expression expression = TestsUtils.parse("[1,2,3,4,5]");

        assertNotNull(expression);
        assertSame(ArrayExpression.class, expression.getClass());

        ArrayExpression arrayExpression = (ArrayExpression) expression;
        assertEquals(5, arrayExpression.values.size());

        for (int i = 0; i < arrayExpression.values.size(); i++) {
            Expression value = arrayExpression.values.get(i);
            assertSame(ValueExpression.class, value.getClass());

            ValueExpression constantValue = (ValueExpression) value;
            assertEquals(i+1, constantValue.value);
        }
    }
}